Automattic/wp-calypso
# Odyssey Widget: add activate buttons when modules are not active

Related to #75923 

## Proposed Changes

The PR proposes to add buttons / message to align with the old widget which we only show for admin users. For non-admin users, the data would be empty (-).

There are some special logic for the transformation of the API return, as protect and akismet returns different format.

## Testing Instructions
* Open your local Jetpack site.
* Check out the branch of the Jetpack to the branch in https://github.com/Automattic/jetpack/pull/30147.
* Build Jetpack if necessary: `jetpack build plugins/jetpack`.
* Run `STATS_PACKAGE_PATH=/path/to/jetpack/projects/packages/stats-admin yarn dev` under `calypso/apps/odyssey-stats`.

* Disable `Brute force protection` from  `/wp-admin/admin.php?page=jetpack#/settings?term=protect`
* Open `/wp-admin/admin.php?odyssey_widget=1`
* Ensure Protect card shows activate button for admin
* Click the button
* Ensure protect module is enabled
* Ensure Protect card shows a number >=0
* Deactivate Aismet plugin
* Open `/wp-admin/index.php?odyssey_widget=1`
* Ensure Akismet card shows a button to activate
* Click the button
* Ensure Akismet is activated
* Uninstall AKismet plugin
* Ensure Akismet card shows a link to Akismet website
* Install and activate Akismet
* Ensure Akismet card shows a number

<img width="536" alt="image" src="https://user-images.githubusercontent.com/1425433/233254806-66fe1041-2520-4a1d-a29f-d1b33112ceaa.png">

<img width="452" alt="image" src="https://user-images.githubusercontent.com/1425433/233261776-07e1661c-e7b7-4f48-92ef-929c4783f701.png">

<img width="598" alt="image" src="https://user-images.githubusercontent.com/1425433/233497969-4a6aacf4-6a58-49dc-89f4-e161113f0733.png">


## Pre-merge Checklist
